File: src/test/java/WordCounterTest.java
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class WordCounterTest {
    private WordCounter wordCounter = new WordCounter();
    @BeforeEach


    @Test
    public void testCountWords() {
        int count = wordCounter.countWords("word wor3d");
        assertEquals(3, count);
    }

    @Test
    public void testCountWordsEmpty() {
        int count = wordCounter.countWords("");
        assertEquals(0, count);
    }

    @Test
    public void testCountWordsMoreNumber() {
        int count = wordCounter.countWords("word wor3222d");
        assertEquals(3, count);
    }

    @Test
    public void testCountWordsJustDelimiters() {
        int count = wordCounter.countWords("223");
        assertEquals(0, count);
    }

    @Test
    public void testStopWords() {
        List<String> stopWords = List.of("the", "a", "an", "off");
        String input = "Mary had a little lamb";
        int count = wordCounter.countWords(input, stopWords);
        assertEquals(4, count);
    }

}

File: src/test/kotlin/WordCounterTest.kt
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.CsvSource

class WordCounterTest {

    val stopWords: List<String> = listOf(
        "the",
        "a",
        "on",
        "off",
    )
    val tested = WordCounter(stopWords)

    @ParameterizedTest
    @CsvSource(
        "'Mary had aa little lamb', 5",
        "'Mary . had , a % little lamb', 4",
        "'Mary 3  had 4  a  little lamb', 4",
        "'M4ry 3  had 4  a  little lamb', 5",
        "'', 0",
        "'the a on off NotATheOnOff', 1",
        "'1 3 . @ ', 0",
    )
    fun countWordsTest(input: String, expectedCount: Int) {
        val result = tested.countWords(input)

        Assertions.assertEquals(expectedCount, result)
    }

}